Formula/etcd-cpp-apiv3.rb

```diff
@@ -0,0 +1,109 @@
+class EtcdCppApiv3 < Formula
+  desc "C++ implementation for etcd's v3 client API, i.e., ETCDCTL_API=3"
+  homepage "https://github.com/etcd-cpp-apiv3/etcd-cpp-apiv3"
+  url "https://github.com/etcd-cpp-apiv3/etcd-cpp-apiv3/archive/refs/tags/v0.2.3.tar.gz"
+  sha256 "f74607eebffce8391b5355ed7d12d46f6627f2882687cd0f5574c90b275f77fd"
+  license "BSD-3-Clause"
+
+  depends_on "cmake" => :build
+  depends_on "etcd" => :test
+
+  depends_on "boost"
+  depends_on "cpprestsdk"
+  depends_on "grpc"
+  depends_on "openssl@1.1"
+  depends_on "protobuf"
+
+  # grpc requrires high version of gcc and are built with high version of gcc,
+  # thus gcc 5 won't work
+  on_linux do
+    depends_on "gcc"
+  end
+
+  fails_with gcc: "5"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build",
+                    "-DCMAKE_CXX_STANDARD=17",
+                    "-DCMAKE_CXX_STANDARD_REQUIRED=TRUE",
+                    "-DBUILD_ETCD_TESTS=OFF",
+                    "-DOPENSSL_ROOT_DIR=#{Formula["openssl@1.1"].opt_prefix}",
+                    *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    # `test do` will create, run in and delete a temporary directory.
+    #
+    # This test will fail and we won't accept that! For Homebrew/homebrew-core
+    # this will need to be a test that verifies the functionality of the
+    # software. Run the test with `brew test etcd-cpp-apiv3`. Options passed
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
+    (testpath/"test.cc").write <<~EOS
+      #include <iostream>
+      #include <etcd/Client.hpp>
+
+      int main() {
+        etcd::Client etcd("http://127.0.0.1:2379");
```

Comment:
can we use `free_port` here to pick a random free port?

Reply:
Fixed.

Reply:
> can we use `free_port` here to pick a random free port?

Fixed.